api/src/test/java/org/apache/iceberg/expressions/TestPredicateBinding.java

```diff
@@ -605,4 +605,188 @@ public void testNotInPredicateBindingConversionToExpression() {
     Expression expr = unbound.bind(struct);
     Assert.assertEquals("Should change NOT_IN to alwaysTrue expression", Expressions.alwaysTrue(), expr);
   }
+
+  @Test
+  public void testStructFields() {
+    StructType structFieldType =
+        Types.StructType.of(Types.NestedField.optional(11, "float_field", Types.FloatType.get()));
+    StructType struct = StructType.of(
+        optional(10, "struct", structFieldType)
+    );
+
+    UnboundPredicate<Float> lt = new UnboundPredicate<>(LT, ref("struct"), 1.234f);
+    try {
+      lt.bind(struct);
```

Comment:
For this and other instance of validation exception, we can use "AssertHelpers.assertThrowsCause"

Reply:
Changed. Thanks!